standalone-metastore/metastore-server/src/main/sql/derby/upgrade-4.0.0-alpha-2-to-4.0.0-beta-1.derby.sql

```diff
@@ -31,18 +31,19 @@ ALTER TABLE "APP"."METASTORE_DB_PROPERTIES" ADD PROPERTYCONTENT BLOB;
 
 -- HIVE-27457
 UPDATE "SDS"
-    SET "SDS"."INPUT_FORMAT" = 'org.apache.hadoop.hive.kudu.KuduInputFormat',
-        "SDS"."OUTPUT_FORMAT" = 'org.apache.hadoop.hive.kudu.KuduOutputFormat'
-    WHERE "SDS"."SD_ID" IN (
-        SELECT "TBL_ID" FROM "TABLE_PARAMS" WHERE "PARAM_VALUE" LIKE '%KuduStorageHandler%'
-    );
+    SET "INPUT_FORMAT" = 'org.apache.hadoop.hive.kudu.KuduInputFormat', "OUTPUT_FORMAT" = 'org.apache.hadoop.hive.kudu.KuduOutputFormat'
+    WHERE "SD_ID" IN (
+        SELECT "TBLS"."SD_ID" FROM "TBLS"
+            INNER JOIN "TABLE_PARAMS" ON "TBLS"."TBL_ID" = "TABLE_PARAMS"."TBL_ID"
+                WHERE "TABLE_PARAMS"."PARAM_VALUE" LIKE '%KuduStorageHandler%'
+        );
 UPDATE "SERDES"
     SET "SERDES"."SLIB" = 'org.apache.hadoop.hive.kudu.KuduSerDe'
     WHERE "SERDE_ID" IN (
         SELECT "SDS"."SERDE_ID"
             FROM "TBLS"
-            LEFT JOIN "SDS" ON "TBLS"."SD_ID" = "SDS"."SD_ID"
-            WHERE "TBL_ID" IN (SELECT "TBL_ID" FROM "TABLE_PARAMS" WHERE "PARAM_VALUE" LIKE '%KuduStorageHandler%')
+            INNER JOIN "SDS" ON "TBLS"."SD_ID" = "SDS"."SD_ID"
```

Comment:
Are you sure the INNER JOIN is better than then LEFT JOIN? The result set could be different. I just want to be curious about it :)

Reply:
I think so, if it is the left join, the result "SDS"."SERDE_ID" could be null if there is no match(though it's impossible), and the `SERDE_ID` is the primary key of `SERDES` , it's always be a non-null long value.